src/command_modules/azure-cli-interactive/azclishell/az_completer.py

```diff
@@ -130,11 +130,21 @@ def __init__(self, commands, global_params=True, outstream=sys.stderr):
 
     def validate_completion(self, param, words, text_before_cursor, double=True):
         """ validates that a param should be completed """
-        return param.lower().startswith(words.lower()) and param.lower() != words.lower() and\
-            param not in text_before_cursor.split() and not \
-            text_before_cursor[-1].isspace() and\
-            (not (double and param in self.same_param_doubles) or
-             self.same_param_doubles[param] not in text_before_cursor.split())
+        # validates the position of the parameter
+        position = param.lower().startswith(words.lower()) and not text_before_cursor[-1].isspace()
+        # cancels parameters that are already in the in line
+        canceling_positions = param.lower() != words.lower() and param not in text_before_cursor.split()
+
+        check_doubles = True
+        # checks for aliasing of parameters
+        if double and self.same_param_doubles.get(self.curr_command):
+            for double_sets in self.same_param_doubles[self.curr_command]:
+                if param in double_sets:
+                    for alias in double_sets:
+                        if alias in text_before_cursor.split() and alias != param:
+                            check_doubles = False
```

Comment:
Ugr, this is a really deep embedded, conditional statement decorated, headache induced loops in loops in loops. Is there a way to simplify the code and make it readable?